airflow-core/src/airflow/migrations/versions/0049_3_0_0_remove_pickled_data_from_xcom_table.py

```diff
@@ -114,12 +114,20 @@ def upgrade():
 
     # Update the values from nan to nan string
     if dialect == "postgresql":
-        # Replace standalone NaN tokens only (not NaN inside string values)
-        # Use regex with word boundaries to match only standalone NaN tokens
+        # Replace NaN in JSON value positions (after :, , or [)
+        # This explicitly matches JSON structure, not relying on word boundaries
         conn.execute(
             text("""
                 UPDATE xcom
-                SET value = convert_to(regexp_replace(convert_from(value, 'UTF8'), '\\bNaN\\b', '"nan"'), 'UTF8')
+                SET value = convert_to(
+                    regexp_replace(
+                        convert_from(value, 'UTF8'),
+                        '([:,\\[])\\s*NaN\\s*([,}\\]])',
```

Comment:
Nit (I think it's a nit, not 100% sure): we are dropping the space here.

```
                        '([:,\\[]\\s*)NaN(\\s*[,}\\]])',
```

would preserve the spaces.

Reply:
Yep, tried and it works well.